src/adapter/src/catalog.rs

```diff
@@ -5995,6 +6028,79 @@ impl Catalog {
                         ObjectId::Role(_) | ObjectId::ClusterReplica(_) => {}
                     }
                 }
+                Op::RenameCluster { id, name, to_name } => {
+                    if id.is_system() {
+                        return Err(AdapterError::Catalog(Error::new(
+                            ErrorKind::ReadOnlyCluster(name.clone()),
+                        )));
+                    }
+                    if is_reserved_name(&to_name) {
+                        return Err(AdapterError::Catalog(Error::new(
+                            ErrorKind::ReservedClusterName(to_name),
+                        )));
+                    }
+                    tx.rename_user_cluster(id, &name, &to_name)?;
+                    builtin_table_updates.push(state.pack_cluster_update(&name, -1));
+                    state.rename_cluster(id, to_name.clone());
+                    builtin_table_updates.push(state.pack_cluster_update(&to_name, 1));
+                    state.add_to_audit_log(
+                        oracle_write_ts,
+                        session,
+                        tx,
+                        builtin_table_updates,
+                        audit_events,
+                        EventType::Alter,
+                        ObjectType::Cluster,
+                        // TODO: Is this the correct EventDetails variant?
```

Comment:
I think you want `RenameItemV1`.

Can you also add a test to the audit_log.slt file that exercises the audit log output for cluster and cluster replica renaming?

Reply:
Ah good point. `RenameCluster` sounds right. I don't think there's enough generality to pursue `RenameObject` and this approach is only a little bit more typing.